app/lib/common/widgets/spaces/space_parent_badge.dart

```diff
@@ -51,9 +52,10 @@ class SpaceParentBadge extends ConsumerWidget {
                     message: parent.profile.displayName,
                     child: InkWell(
                       onTap: onTap ??
-                          () {
-                            context.push('/$parentId');
-                          },
+                          () => context.goNamed(
+                                Routes.space.name,
+                                pathParameters: {'spaceId': parentId},
+                              ),
```

Comment:
but that won't work for the other cases where this widget is used. Namely, if you clicked this from a chat now, it wouldn't push the route and you wouldn't have any `back` on top ...

I suggest we create another param `this.push = true` that the surrounding part can set to `false` (like in the space header) that switches from `pushNamed` to `goNamed` behaviour.

Reply:
How we would determine if this is a push ?? Even if we do, it would still make shell disappear as ```push``` is executed in either case. The problem lies how the ShellRoute works with pushed route unfortunately because of the [issue](https://github.com/flutter/flutter/issues/120665)...